src/lib/comms/sol-coap.c

```diff
@@ -1072,7 +1072,7 @@ sol_coap_packet_new_request(sol_coap_method_t method, sol_coap_msgtype_t type)
 }
 
 SOL_API int
-sol_coap_add_option(struct sol_coap_packet *pkt, uint16_t code, const void *value, uint16_t len)
+sol_coap_add_option(struct sol_coap_packet *pkt, uint16_t code, const void *value, size_t len)
```

Comment:
The use of `uint16_t` documents that the option format only allows an option length that can be represented in 16 bits (the actual max would be 65804 IIRC, but I didn't see much value in supporting that).

If you really want to change this, also update `coap_option_encode()` and `coap_parse_option()` to take this into account.